Document the default build environment variables (#4496)
ocaml/opam: doc/pages/Manual.md

- <a id="opamfield-build-env">`build-env: [ <environment-update> ... ]`</a>:
  defines environment updates that will be applied when running the package's
  build, install and remove scripts.
  

>   By default, the following environment variables are set by opam:

Reviewer: ```suggestion
  The following environment variables are set by opam (but can be overridden by `build-env`):
```

  + `CDPATH=`
  + `MAKEFLAGS=`
  + `MAKELEVEL=`

>   + `OPAM_PACKAGE_NAME=<pkg>` (with `<pkg>` being the name of the current package being built/installed/removed)
>   + `OPAM_PACKAGE_VERSION=<ver>` (with `<ver>` being the version of the current package being built/installed/removed)

Reviewer: ```suggestion
  + `OPAM_PACKAGE_NAME=<pkg>` (`<pkg>` is the name of the package being built/installed/removed)
  + `OPAM_PACKAGE_VERSION=<ver>` (`<ver>` is the version of the package being built/installed/removed)
```
(slightly more terse)

  + `OPAMCLI=2.0` (since opam 2.1)

>   + `TMP` and `TMPDIR` are also set by the sandbox (bubblewrap) script but are not to be relied upon.

Reviewer: I'd reword this slightly to explain why:
```suggestion
  + `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since ...
```

Author: ```suggestion
  + `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on every platforms and can be disabled by the user.
```
What about this?

Reviewer: LGTM!

  defines environment updates that will be applied when running the package's
  build, install and remove scripts.

    The following environment variables are set by opam (but can be overridden by `build-env`):
    - `CDPATH=`
    - `MAKEFLAGS=`
    - `MAKELEVEL=`
    - `OPAM_PACKAGE_NAME=<pkg>` (`<pkg>` is the name of the package being built/installed/removed)
    - `OPAM_PACKAGE_VERSION=<ver>` (`<ver>` is the version of the package being built/installed/removed)
    - `OPAMCLI=2.0` (since opam 2.1)

>     - `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on every platforms and can be disabled by the user.

Reviewer: ```suggestion
    - `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on all platforms and can be disabled by the user.
```
